syntax = "proto3";

// Run `protoc --python_out=temp doorbell.proto`

package eufy_security.mqtt;

message NotImplemented {
}

message NotificationData {
  string picURL = 1;
  string videoURL = 2;
  int32 monitorId = 3;
  int32 cipher = 4;
  uint64 createTime = 5;
  int32 channel = 6;
  string filePath = 7;
  string stationSN = 10;
  string deviceSN = 11;
  int32 notificationStyle = 12;
  int32 pushCount = 17;
  int32 pushJump = 18;
}

message DeviceSettingNotify {
  uint64 timestamp = 1;
  string sessionId = 2;
  bool boolValue = 10;
  int32 intValue = 11;
  NotImplemented activeZone = 16;
  NotImplemented sdCardState = 18;
  NotImplemented OTAData = 19;
  NotificationData notificationData = 20;
  NotImplemented motionDetectionMode = 21;
  NotImplemented historyDeleteData = 22;
  NotImplemented quickRespData = 23;
  NotImplemented json = 24;
}

enum MessageType {
  UNDEFINE_SETTING_MESSAGE_VALUE = 0;
  NOTIFY_CHIME_BIND_VALUE = 3016;
  NOTIFY_CHIME_OPEN_VALUE = 3015;
  NOTIFY_DELETE_HISTORY_EVENT_VALUE = 3014;
  NOTIFY_DEVICE_OPEN_VALUE = 3001;
  NOTIFY_DEVICE_RINGING_VOLUME_VALUE = 3022;
  NOTIFY_DEVICE_VOLTAGE_VALUE = 3021;
  NOTIFY_DEVICE_VOLUME_VALUE = 3003;
  NOTIFY_EVENT_BACKGROUND_ACTIVE_VALUE = 3100;
  NOTIFY_EVENT_FACE_DETECTION_VALUE = 3102;
  NOTIFY_EVENT_MOTION_DETECTION_VALUE = 3101;
  NOTIFY_EVENT_PRESS_DOORBELL_VALUE = 3103;
  NOTIFY_EVENY_FACE_DETECTION_OVER_VALUE = 3122;
  NOTIFY_EVENY_MOTION_DETECTION_OVER_VALUE = 3121;
  NOTIFY_EVENY_PRESS_DOORBELL_OVER_VALUE = 3123;
  NOTIFY_FIRMWARE_UPDATE_VALUE = 3012;
  NOTIFY_FORMAT_SDCARD_COMPLETED_VALUE = 3011;
  NOTIFY_METERING_ZONE_DELETE_VALUE = 3024;
  NOTIFY_METERING_ZONE_VALUE = 3023;
  NOTIFY_MODE_ACTION_CONFIG_VALUE = 3009;
  NOTIFY_MODE_CONFIG_VALUE = 3008;
  NOTIFY_MODE_SIWTCH_VALUE = 3018;
  NOTIFY_MOTION_DETECTION_LEVEL_VALUE = 3005;
  NOTIFY_MOTION_DETECTION_MODE_VALUE = 3004;
  NOTIFY_MOTION_DETECTION_SCENARIO_VALUE = 3025;
  NOTIFY_MOTION_DETECTION_ZONE_DELETE_VALUE = 3007;
  NOTIFY_MOTION_DETECTION_ZONE_VALUE = 3006;
  NOTIFY_NIGHTVISUAL_OPEN_VALUE = 3002;
  NOTIFY_SCHEDULE_CONFIG_VALUE = 3019;
  NOTIFY_SCHEDULE_DEFAULT_MODE_VALUE = 3020;
  NOTIFY_SDCARD_STATE_VALUE = 3010;
  NOTIFY_UNBIND_DEVICE_VALUE = 3013;
  NOTIFY_UPDATE_QUICK_RESPONSE_VALUE = 3017;
  REQUEST_CHIME_BIND_VALUE = 1016;
  REQUEST_CHIME_OPEN_VALUE = 1015;
  REQUEST_DELETE_HISTORY_EVENT_VALUE = 1014;
  REQUEST_DEVICE_OPEN_VALUE = 1001;
  REQUEST_DEVICE_RINGING_VOLUME_VALUE = 1022;
  REQUEST_DEVICE_VOLTAGE_VALUE = 1021;
  REQUEST_DEVICE_VOLUME_VALUE = 1003;
  REQUEST_FIRMWARE_UPDATE_VALUE = 1012;
  REQUEST_FORMAT_SDCARD_VALUE = 1011;
  REQUEST_METERING_ZONE_DELETE_VALUE = 1024;
  REQUEST_METERING_ZONE_VALUE = 1023;
  REQUEST_MODE_ACTION_CONFIG_VALUE = 1009;
  REQUEST_MODE_CONFIG_VALUE = 1008;
  REQUEST_MODE_SIWTCH_VALUE = 1018;
  REQUEST_MOTION_DETECTION_LEVEL_VALUE = 1005;
  REQUEST_MOTION_DETECTION_MODE_VALUE = 1004;
  REQUEST_MOTION_DETECTION_SCENARIO_VALUE = 1025;
  REQUEST_MOTION_DETECTION_ZONE_DELETE_VALUE = 1007;
  REQUEST_MOTION_DETECTION_ZONE_VALUE = 1006;
  REQUEST_NIGHTVISUAL_OPEN_VALUE = 1002;
  REQUEST_PUSH_CAM_INFO_VALUE = 1110;
  REQUEST_PUSH_CAM_IRCUT_VALUE = 1112;
  REQUEST_PUSH_CAM_IRLED_VALUE = 1111;
  REQUEST_PUSH_MEDIA_STREAM_START_VALUE = 1100;
  REQUEST_PUSH_MEDIA_STREAM_STOP_VALUE = 1101;
  REQUEST_PUSH_OTA_START_VALUE = 1102;
  REQUEST_PUSH_REBOOT_VALUE = 1104;
  REQUEST_PUSH_UPLOAD_LOG_VALUE = 1103;
  REQUEST_SCHEDULE_CONFIG_VALUE = 1019;
  REQUEST_SCHEDULE_DEFAULT_MODE_VALUE = 1020;
  REQUEST_SDCARD_STATE_VALUE = 1010;
  REQUEST_UNBIND_DEVICE_VALUE = 1013;
  REQUEST_UPDATE_QUICK_RESPONSE_VALUE = 1017;
  RESPONSE_CHIME_BIND_VALUE = 2016;
  RESPONSE_CHIME_OPEN_VALUE = 2015;
  RESPONSE_DELETE_HISTORY_EVENT_VALUE = 2014;
  RESPONSE_DEVICE_OPEN_VALUE = 2001;
  RESPONSE_DEVICE_RINGING_VOLUME_VALUE = 2022;
  RESPONSE_DEVICE_VOLTAGE_VALUE = 2021;
  RESPONSE_DEVICE_VOLUME_VALUE = 2003;
  RESPONSE_FIRMWARE_UPDATE_VALUE = 2012;
  RESPONSE_FORMAT_SDCARD_VALUE = 2011;
  RESPONSE_METERING_ZONE_DELETE_VALUE = 2024;
  RESPONSE_METERING_ZONE_VALUE = 2023;
  RESPONSE_MODE_ACTION_CONFIG_VALUE = 2009;
  RESPONSE_MODE_CONFIG_VALUE = 2008;
  RESPONSE_MODE_SIWTCH_VALUE = 2018;
  RESPONSE_MOTION_DETECTION_LEVEL_VALUE = 2005;
  RESPONSE_MOTION_DETECTION_MODE_VALUE = 2004;
  RESPONSE_MOTION_DETECTION_SCENARIO_VALUE = 2025;
  RESPONSE_MOTION_DETECTION_ZONE_DELETE_VALUE = 2007;
  RESPONSE_MOTION_DETECTION_ZONE_VALUE = 2006;
  RESPONSE_NIGHTVISUAL_OPEN_VALUE = 2002;
  RESPONSE_PUSH_CAM_INFO_VALUE = 2110;
  RESPONSE_PUSH_CAM_IRCUT_VALUE = 2112;
  RESPONSE_PUSH_CAM_IRLED_VALUE = 2111;
  RESPONSE_PUSH_MEDIA_STREAM_START_VALUE = 2100;
  RESPONSE_PUSH_MEDIA_STREAM_STOP_VALUE = 2101;
  RESPONSE_PUSH_OTA_START_VALUE = 2102;
  RESPONSE_PUSH_REBOOT_VALUE = 2104;
  RESPONSE_PUSH_UPLOAD_LOG_VALUE = 2103;
  RESPONSE_SCHEDULE_CONFIG_VALUE = 2019;
  RESPONSE_SCHEDULE_DEFAULT_MODE_VALUE = 2020;
  RESPONSE_SDCARD_STATE_VALUE = 2010;
  RESPONSE_UNBIND_DEVICE_VALUE = 2013;
  RESPONSE_UPDATE_QUICK_RESPONSE_VALUE = 2017;
}

message DeviceSettingMessage {
  MessageType msgType = 1;
  int32 msgVer = 2;
  string userId = 3;
  NotImplemented request = 13;
  NotImplemented response = 14;
  DeviceSettingNotify deviceSettingNotify = 15;
}

